src/ocean/io/serialize/StructSerializer.d

```diff
@@ -575,12 +575,8 @@ struct StructSerializer ( bool AllowUnions = false )
 
                 static if (isTypedef!(T))
                 {
-                    mixin(`
-                    static if ( is(T B == typedef) )
-                    {
-                        serializer.serialize(data, cast(B)(field), field_name);
-                    }
-                    `);
+                    serializer.serialize(data, cast(StripTypedef!(T))(field),
```

Comment:
ABI **has** to be compatible. If not, D2 apps won't be able to deserialize profiles that contain `typedef` which were serialized in D1.